pkg/utils/tls/tls_test.go

```diff
@@ -0,0 +1,206 @@
+// Copyright 2025 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tls
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"math/big"
+	"os"
+	"testing"
+	"time"
+)
+
+func generateTestCert(t *testing.T, validFor time.Duration) (certFile, keyFile string) {
+	t.Helper()
+
+	certFile = mustTempFile(t)
+	keyFile = mustTempFile(t)
+
+	privateKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		t.Fatalf("Failed to generate private key: %v", err)
+	}
+
+	now := time.Now()
+	template := x509.Certificate{
+		SerialNumber:          big.NewInt(1),
+		Subject:               pkix.Name{CommonName: "test-cert"},
+		NotBefore:             now,
+		NotAfter:              now.Add(validFor),
+		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+		BasicConstraintsValid: true,
+	}
+
+	certDER, err := x509.CreateCertificate(rand.Reader, &template, &template, &privateKey.PublicKey, privateKey)
+	if err != nil {
+		t.Fatalf("Failed to create certificate: %v", err)
+	}
+
+	certOut, err := os.Create(certFile)
+	if err != nil {
+		t.Fatalf("Failed to create cert file: %v", err)
+	}
+	defer certOut.Close()
+	pem.Encode(certOut, &pem.Block{Type: "CERTIFICATE", Bytes: certDER})
+
+	keyOut, err := os.Create(keyFile)
+	if err != nil {
+		t.Fatalf("Failed to create key file: %v", err)
+	}
+	defer keyOut.Close()
+	pem.Encode(keyOut, &pem.Block{
+		Type:  "RSA PRIVATE KEY",
+		Bytes: x509.MarshalPKCS1PrivateKey(privateKey),
+	})
+
+	return certFile, keyFile
+}
+
+func TestLoadTLSCert(t *testing.T) {
```

Comment:
TBH I'm not sure if it makes sense to test this function. It's a wrapper around `tls.LoadX509KeyPair`. I'd say the main purpose of this function is to provide a clearer error message, so IMHO we should test the following two scenarios:
- A good certificate, just to be sure we don't mess up so bad with this one
- Check that the `loading TLS keypair` string is contained in the error returned when parsing a bad (non existing) certificate